### PR TITLE
IGNITE-14030 Adds containKeys, clearKey, clearKeys and getAndPutIfAbsent operations to Java client API.

### PR DESCRIPTION
**Pull request by ololo3000**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/client/ClientCache.java [hunk 1]
```diff
@@ -558,6 +575,47 @@
...
+     * with the specified key and a value was set.
+     */
+    public V getAndPutIfAbsent(K key, V val) throws ClientException;
+
+    /**
+     * Atomically associates the specified key with the given value if it is not already associated with a value.
+     * <p>
+     * This is equivalent to performing the following operations as a single atomic action:
+     * <pre><code>
+     * if (!cache.containsKey(key)) {}
```
[the remark this answers is not in the document]
Comment (ololo3000) [reply modules/core/src/main/java/org/apache/ignite/client/ClientCache.java]: Done.

File: modules/core/src/main/java/org/apache/ignite/client/ClientCache.java [hunk 2]
```diff
@@ -571,6 +629,32 @@
      */
     public IgniteClientFuture<Void> clearAsync() throws ClientException;
 
+    /**
+     * Clears entry with specified key from the cache.
+     * In contrast to {@link #remove(Object)}}, this method does not notify event listeners and cache writers.
+     */
+    public void clear(K key);
```
[the remark this answers is not in the document]
Comment (ololo3000) [reply modules/core/src/main/java/org/apache/ignite/client/ClientCache.java]: Done.

File: modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientCache.java [hunk 3]
```diff
@@ -570,6 +600,38 @@
...
+        return cacheSingleKeyOperation(
+            key,
+            ClientOperation.CACHE_GET_AND_PUT_IF_ABSENT,
+            req -> writeObject(req, val),
+            this::readObject
+        );
+    }
+
+    /** {@inheritDoc} */
+    @Override public IgniteClientFuture<V> getAndPutIfAbsentAsync(K key, V val) throws CacheException, TransactionException {
```
[the remark this answers is not in the document]
Comment (ololo3000) [reply modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientCache.java]: Done.

File: modules/core/src/main/java/org/apache/ignite/client/ClientCache.java [hunk 4]
```diff
@@ -571,6 +629,32 @@
...
+
+    /**
+     * Clears entry with specified key from the cache asynchronously.
+     * In contrast to {@link #removeAsync(Object)}}, this method does not notify event listeners and cache writers.
```
[the remark this answers is not in the document]
Comment (ololo3000) [reply modules/core/src/main/java/org/apache/ignite/client/ClientCache.java:640]: Done.

File: modules/core/src/main/java/org/apache/ignite/client/ClientCache.java [hunk 5]
```diff
@@ -571,6 +629,32 @@
      */
     public IgniteClientFuture<Void> clearAsync() throws ClientException;
 
+    /**
+     * Clears entry with specified key from the cache.
+     * In contrast to {@link #remove(Object)}}, this method does not notify event listeners and cache writers.
```
[the remark this answers is not in the document]
Comment (ololo3000) [reply modules/core/src/main/java/org/apache/ignite/client/ClientCache.java:634]: Done.

File: modules/core/src/main/java/org/apache/ignite/client/ClientCache.java [hunk 6]
```diff
@@ -558,6 +575,47 @@
      */
     public IgniteClientFuture<Boolean> putIfAbsentAsync(K key, V val) throws ClientException;
 
+    /**
+     * Atomically associates the specified key with the given value if it is not already associated with a value.
+     * <p>
+     * This is equivalent to performing the following operations as a single atomic action:
+     * <pre><code>
+     * if (!cache.containsKey(key)) {}
```
[the remark this answers is not in the document]
Comment (ololo3000) [reply modules/core/src/main/java/org/apache/ignite/client/ClientCache.java]: Done.

File: modules/core/src/main/java/org/apache/ignite/client/ClientCache.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (alex-plekhanov) [reply modules/core/src/main/java/org/apache/ignite/client/ClientCache.java:634]: Still not fixed `{@link #remove(Object)}**}**`

[the remark this answers is not in the document]
Comment (ololo3000) [reply modules/core/src/main/java/org/apache/ignite/client/ClientCache.java]: Sorry,  I spoke too soon. Done.

**Review by alex-plekhanov — APPROVED** [review]

**Pull request closed without merging**